csharp/Adapter/Microsoft.Spark.CSharp/Services/LoggerServiceFactory.cs

```diff
@@ -11,17 +12,15 @@ namespace Microsoft.Spark.CSharp.Services
     /// </summary>
     public class LoggerServiceFactory
     {
-        private static ILoggerService loggerService = DefaultLoggerService.Instance;
-        
+        private static Lazy<ILoggerService> loggerService = new Lazy<ILoggerService>(() => GetDefaultLogger());
+
         /// <summary>
         /// Overrides an existing logger by a given logger service instance
         /// </summary>
         /// <param name="loggerServiceOverride">The logger service instance used to overrides</param>
         public static void SetLoggerService(ILoggerService loggerServiceOverride)
```

Comment:
Refer to the usage of this method in https://github.com/Microsoft/Mobius/blob/master/examples/Batch/WordCount/Program.cs#L21. Make sure that we are compatible with the old behavior.